magda-storage-api/src/test/storage.spec.ts

```diff
@@ -236,47 +261,147 @@ describe("Storage API tests", () => {
+                )
+                .reply(404);
+        });
+
+        describe("Uploading and then downloading a simple file", () => {
+            it("With registry auth", () => {
+                console.log("running test 1");
+                return mockAuthorization(
+                    authApiUrl,
+                    true,
+                    jwtSecret,
+                    request(app)
+                        .put(
+                            "/v0/" +
+                                bucketName +
+                                "/download-registry-auth-test-file-1"
+                        )
+                        .set("Accept", "application/json")
+                        .set("Content-Type", "text/plain")
+                        .query({ recordId: "storage-test-dataset-id-success" })
+                        .send("Testing download")
+                        .expect(200)
+                ).then(_res => {
+                    return injectUserId(
+                        jwtSecret,
+                        request(app)
+                            .get(
+                                "/v0/" +
+                                    bucketName +
+                                    "/download-registry-auth-test-file-1"
+                            )
+                            .set("Accept", "application/json")
+                            .set("Accept", "text/plain")
+                            .expect(200)
+                            .expect("Testing download")
+                            .expect("Content-Type", "text/plain")
+                    );
+                });
+            });
+
+            it("Without registry auth", () => {
+                console.log("running test 2");
+                return mockAuthorization(
+                    authApiUrl,
+                    true,
+                    jwtSecret,
+                    request(app)
+                        .put(
+                            "/v0/" + bucketName + "/download-unauth-test-file-1"
+                        )
+                        .set("Accept", "application/json")
+                        .set("Content-Type", "text/plain")
+                        .query({ recordId: "storage-test-dataset-id-failure" })
+                        .send("Testing download without auth")
+                        .expect(200)
+                ).then(_res => {
+                    return injectUserId(
+                        jwtSecret,
+                        request(app)
+                            .get(
+                                "/v0/" +
+                                    bucketName +
+                                    "/download-unauth-test-file-1"
+                            )
+                            .set("Accept", "application/json")
+                            .set("Accept", "text/plain")
+                            .expect(404)
+                    );
+                });
+            });
+        });
```

Comment:
It'd be good to have a test for what happens when the file doesn't have a record id attached (upload and download) as well.

Reply:
Added